src/transformers/modeling_utils.py

```diff
@@ -2251,10 +2280,19 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                 torch_dtype = torch.float16
 
             if device_map is None:
-                raise ValueError(
-                    "A device map needs to be passed to run convert models into 8-bit and 4-bit formats. Please run"
-                    "`.from_pretrained` with `device_map='auto'`"
+                if torch.cuda.is_available():
+                    current_device = cls._get_current_device()
```

Comment:
Let's use the torch cuda device as before. If it's not the right one, then the user can pass along the `device_map`.